mlir/lib/Quantum/Transforms/remove_chained_self_inverse.cpp

```diff
@@ -48,9 +50,24 @@ struct RemoveChainedSelfInversePass
         LLVM_DEBUG(dbgs() << "remove chained self inverse pass"
                           << "\n");
 
+        Operation *module = getOperation();
+        Operation *targetfunc;
+
+        module->walk([&](Operation *op) {
+            if (isa<func::FuncOp>(op)) {
+                StringRef funcName = cast<func::FuncOp>(op).getSymName();
+
+                if (funcName != FuncNameOpt) {
+                    return; // not the function to run the pass on, so do nothing and exit
+                }
+                targetfunc = op;
+            }
+        });
```

Comment:
Same comment as before. 

Tangential: I originally thought that it would be better to supply multiple function names instead of a single one. I am not so convinced right now that supplying a list would be any better. If we schedule this transformation using `func.func(cancel-inverses)` we should only match a single one. But if it is on the module level... I'm still thinking about this... :thinking: Will give this a bit more thought.

Reply:
I think this is only a distinction between "flattening" the list in the `apply-transform-pattern` overall scheduler or in the `remove-inverse` pass itself; overall neither of these options should have a significance advantage over the other. 

Also, because the frontend decorator is applied to single qnodes, if the passes were to take in lists then the frontend would have an extra work of keeping track of the list, only to be unpacked here. In addition, I think the point of the pipeline is to schedule multiple passes for each qnode, instead of scheduling batches of qnode for each pass (at least that's my interpretation of Josh's epic)

Reply:
> I think the point of the pipeline is to schedule multiple passes for each qnode

Yes, you are right. Thanks, I am satisfied with this answer :)